src/Cassandra.IntegrationTests/Core/PoolShortTests.cs

```diff
@@ -239,5 +239,115 @@ public void PoolingOptions_Create_Based_On_Protocol(ProtocolVersion protocolVers
+                Assert.AreEqual(2, connections.Count);
+                await testCluster.DropConnection(connections[0]);
+                await testCluster.DropConnection(connections[1]);
+
+                TestHelper.WaitUntil(() => !cluster.GetHost(ccAddress).IsUp);
+
+                Assert.False(cluster.GetHost(ccAddress).IsUp);
+
+                TestHelper.WaitUntil(() => !cluster.GetControlConnection().Address.Address.Equals(ccAddress.Address));
+
+                Assert.AreNotEqual(ccAddress.Address, cluster.GetControlConnection().Address.Address);
+
+                // Previous host is still DOWN
+                Assert.False(cluster.GetHost(ccAddress).IsUp);
+
+                // New host is UP
+                ccAddress = cluster.GetControlConnection().Address;
+                Assert.True(cluster.GetHost(ccAddress).IsUp);
+            }
+        }
+
+        [Test]
+        public async Task ControlConnection_Should_Reconnect_After_Failed_Attemps()
+        {
+            const int connectionLength = 1;
+            var builder = Cluster.Builder()
+                                 .WithPoolingOptions(new PoolingOptions()
+                                     .SetCoreConnectionsPerHost(HostDistance.Local, connectionLength)
+                                     .SetMaxConnectionsPerHost(HostDistance.Local, connectionLength)
+                                     .SetHeartBeatInterval(1000))
+                                 .WithReconnectionPolicy(new ConstantReconnectionPolicy(100L));
+            using (var testCluster = SimulacronCluster.CreateNew(new SimulacronOptions { Nodes = "3" }))
+            using (var cluster = builder.AddContactPoint(testCluster.InitialContactPoint).Build())
+            {
+                var session = (Session)cluster.Connect();
+                var allHosts = cluster.AllHosts();
+                Assert.AreEqual(3, allHosts.Count);
+                await TestHelper.TimesLimit(() =>
+                    session.ExecuteAsync(new SimpleStatement("SELECT * FROM system.local")), 100, 16);
+
+                var serverConnections = testCluster.GetConnectedPorts();
+                // 1 per hosts + control connection
+                WaitSimulatorConnections(testCluster, 4);
+                Assert.AreEqual(4, serverConnections.Count);
+
+                // Disable all connections
+                await testCluster.DisableConnectionListener();
+
+                var ccAddress = cluster.GetControlConnection().Address;
+
+                // Drop all connections to hosts
+                foreach (var connection in serverConnections)
+                {
+                    await testCluster.DropConnection(connection);
+                }
+
+                TestHelper.WaitUntil(() => !cluster.GetHost(ccAddress).IsUp);
+
+                // All host should be down by now
+                TestHelper.WaitUntil(() => cluster.AllHosts().All(h => !h.IsUp));
```

Comment:
As the HeartBeatInterval its been set, wdyt about also Assert the amount of time it took to mark the node up and down?

Reply:
Its a good idea but I would like to avoid being too specific around time as we the precision is not guaranteed (there are huge difference between runs on AppVeyor specially).